changeset.go

```diff
@@ -871,7 +871,12 @@ func (cs *Changeset) revertDaemonSet(ctx context.Context, item *ChangesetItem) e
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		return control.Delete(ctx, true)
+		err = control.Delete(ctx, true)
+		// If the resource has already been deleted, suppress the error
+		if trace.IsNotFound(err) {
```

Comment:
I wonder if this would not have been more idiomatic to handle on the top-level - to capture all future resource types automatically:
```go
	for i := len(tr.Spec.Items) - 1; i >= 0; i-- {
		op := &tr.Spec.Items[i]
		info, err := GetOperationInfo(*op)
		if err != nil {
			return trace.Wrap(err)
		}
		if op.Status != OpStatusCompleted {
			log.Infof("skipping changeset item %v, status: %v is not the expected %v", info, op.Status, OpStatusCompleted)
		}
		err = cs.revert(ctx, op, info)
		if err != nil && !trace.IsNotFound(err) { // Ignore not found errors during revert
			return trace.Wrap(err)
		}
		...
	}
```

Reply:
Yea maybe, I was a little less comfortable doing so, because I didn't really dig in far enough to know if we generate a NotFound error in any other paths, that would then get accidentally swallowed at the top level.